codelists/tests/test_actions.py

```diff
@@ -781,3 +781,14 @@ def test_clone_codelist_user_own_codelist(user_codelist, organisation_user):
 
     assert user_codelist.name in clone2.name
     assert "clone1" in clone2.name
+
+
+def test_re_clone_organisational_codelist(new_style_codelist, user):
```

Comment:
I tried running this test against the old code, just to check that it would fail, and it did but with a slightly different error message compared to the original sentry error. 

With the failing unit test, I saw:
`django.db.utils.IntegrityError: UNIQUE constraint failed: codelists_handle.user_id, codelists_handle.slug`

However, in Sentry the error is:
`IntegrityError: UNIQUE constraint failed: codelists_handle.user_id, codelists_handle.name`

i.e one is complaining about the slug and one is complaining about the name. Is this definitely solving the same problem?

Reply:
e.g. 

```py
if (
        Handle.objects.filter(user=new_owner, name=name).exists()
        | Handle.objects.filter(user=new_owner, slug=codelist.slug).exists()
    ):
```

Reply:
@lucyb  any thoughts?

Reply:
Thinking about it, it's possible my local database is in a weird state, I've not updated it recently and if you're getting the expected error then it's probably fine.

The extra check is probably fine, although I guess there's a slightly increased risk of an infinite loop here maybe?
```
        n = 0
        while True:
            try:
                new_name = f"{codelist.name} (clone{str(n) if n > 0 else ''})"
                Handle.objects.get(user=new_owner, name=new_name)
            except Handle.DoesNotExist:
                name = new_name
                break
            n += 1
```
I appreciate this hasn't changed in this PR, but would it be worth putting an upper limit on n?

Reply:
Oh sorry, I misread the code! Yes, you're absolutely right. Although, it's probably good that we find out if people have been creating lots of cloned codelists.